ClangSharp.PInvokeGenerator/Cursors/Decls/ClassDecl.cs

```diff
@@ -2,124 +2,11 @@
 
 namespace ClangSharp
 {
-    internal sealed class ClassDecl : Decl
+    internal sealed class ClassDecl : CXXRecordDecl
```

Comment:
A `ClassDecl` doesn't actually exist in the C++ type hierarchy, they are all just `RecordDecl` or `CXXRecordDecl`. I haven't finished investigating if that distinction is important for binding generation.

Reply:
Doesn't look to be important outside of whether it is a Union or Class/Struct.

I've changed it to detect the language type to pick between `RecordDecl` and `CXXRecordDecl` and to expose `IsStruct`, `IsClass`, and `IsUnion` properties.